site/zenodo_rdm/legacy/deserializers/schemas.py

```diff
@@ -38,6 +73,72 @@ def _creators(self, data):
             ret.append(r)
         return ret
 
+    def _funder(self, funder):
+        """Transform the funder information.
+
+        From Legacy Zenodo (invenio-opeanaire) funder
+        (invenio_openaire/jsonschemas/funders/funder-v1.0.0.json) to RDM funder
+        (invenio_vocabularies/contrib/funders/jsonschemas/funders/funder-v1.0.0.json).
+        If the main identifier (DOI to ROR) is not found, it is passed as a custom
+        funder (only name).
+        """
+
+        id_ = funder.get("doi")
+        if not id_:
+            # No id to look ROR for, therefore it is a custom funder
+            return {
+                "name": funder.get("name"),
+            }
+
+        ror = FUNDER_DOI_TO_ROR.get(id_, None)
+        if ror:
+            return {"id": ror}
+
+        return {
+            # special case of funder loaded in Zenodo with no DOI related to it.
+            # it is part of the vocabulary.
+            "id": funder.get("doi"),
+        }
+
+    def _award(self, grant):
+        """Transform the award information.
+
+        From Legacy Zenodo (invenio-opeanaire) grant
+        (invenio_openaire/jsonschemas/grants/grant-v1.0.0.json) to RDM award
+        (invenio_vocabularies/contrib/awards/jsonschemas/awards/award-v1.0.0.json).
+        """
+
+        id_ = grant.get("internal_id")
+        if id_:  # should be part of vocabulary
+            # format `funder_id::code` as per
+            # invenio_vocabularies/contrib/awards/datastreams.py#L53
+            return {"id": id_}
+
+        # custom award
+        return {
+            "title": grant["title"],
+            "number": grant["code"],
+            "identifiers": [
```

Comment:
We might have to configure [VOCABULARIES_AWARD_SCHEMES](https://github.com/inveniosoftware/invenio-vocabularies/blob/master/invenio_vocabularies/config.py#L58). At the momtne it contains only DOI and URL, however legacy records would contain OAF, pURL and eurepo.